src/extrapolation/extrap_prep.jl

```diff
@@ -0,0 +1,138 @@
+"""
```

Comment:
The comments are helpful, but it might be even easier to understand with some kind of overarching comment at the top. Otherwise these help phrases don't mean a lot.